pkg/plugins/client.go

```diff
@@ -0,0 +1,415 @@
+package plugins
+
+import (
+	zipa "archive/zip"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/mod/module"
+	"golang.org/x/mod/zip"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	sourcesFolder  = "sources"
+	archivesFolder = "archives"
+	stateFilename  = "state.json"
+	goPathSrc      = "src"
+	pluginManifest = ".traefik.yml"
+)
+
+const pilotURL = "https://plugin.pilot.traefik.io/public/"
+
+const (
+	hashHeader  = "X-Plugin-Hash"
+	tokenHeader = "X-Token"
+)
+
+// ClientOptions the options of a Traefik Pilot client.
+type ClientOptions struct {
+	Output string
+	Token  string
+}
+
+// Client a Traefik Pilot client.
+type Client struct {
+	HTTPClient *http.Client
```

Comment:
Is there a reason to have this client exposed?

Reply:
yes, it allow to override the HTTP client for testing.